### PR TITLE
add proto for livestream health exports stream

### PR DESCRIPTION
**Pull request by electrachong**
Add the protobuf spec for the livestream health exports stream.

### CODE REVIEW

**Comment by electrachong** [comment]
@jeremycondon and @stevenlyons just added you two in case you have opinions about the file hierarchy / naming in this repo, feel free to disregard if it's not of interest!

**Review by jeremycondon — COMMENTED** [review]
I think this should match the structure of the other export, can you add a basic README.md and put the file in /proto?

**Comment by electrachong** [comment]
Updated to put `livestream-health-exports` under a `video` namespace directory and add the missing `proto` subdirectory & readme

**Comment by electrachong** [comment]
There's still some discussion about the proper namespace for the proto definition so @nidhik will follow up to make updates as needed to this PR while I'm out next week.

File: video/livestream-health-exports/proto/exportstream.proto [hunk 1]
```diff
@@ -0,0 +1,75 @@
+syntax = "proto2";
+
+package exportstream.livehealth;
```
Comment (electrachong) [inline video/livestream-health-exports/proto/exportstream.proto:3]: The package dictates the name of the Ruby gem (if compiled to Ruby). I found that if you load two gems of the same names it seems to merge the classes, so you could access:
```
irb(main):017:0> Exportstream::AudioTrack
=> Exportstream::AudioTrack
irb(main):018:0> Exportstream::ExternalVideoViewRecord
=> Exportstream::ExternalVideoViewRecord
```
but I thought `Exportstream::Livehealth::AudioTrack` might be preferable. I'm happy to leave this up to discretion though.

https://developers.google.com/protocol-buffers/docs/reference/ruby-generated

Comment (electrachong) [reply video/livestream-health-exports/proto/exportstream.proto:3]: An alternative could be:
```
package livehealthexportstream
```
->
```
Livehealthexportstream::AudioTrack
```

File: video/livestream-health-exports/proto/exportstream.proto [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+option go_package = "github.com/muxinc/mux/data/internal/proto/golang/exportstream/livehealth";
+option java_package = "com.mux.data.internal.proto.exportstream.livehealth";
+
+import "google/protobuf/timestamp.proto";
+
+// The generic live health events
+
+message AudioTrack {
```
Comment (jeremycondon) [inline video/livestream-health-exports/proto/exportstream.proto:12]: We have a conversation going about fixing this for view stream exports, but for pubsub to work with this, we need to only have 1 "top-level type"

> For a schema to work with Pub/Sub, only one top-level type may be defined. Import statements to reference other types are also not supported.

File: video/livestream-health-exports/proto/exportstream.proto (hunk 1 above)
Comment (jeremycondon) [reply video/livestream-health-exports/proto/exportstream.proto:3]: not sure if this helps, but looks like we can specify per "language" what the package name should be:
`option ruby_package = "Google::Cloud::PubSub::V1";`
Maybe: `option ruby_package = "Mux::ExportStream::NameHere"`

File: video/livestream-health-exports/proto/exportstream.proto (hunk 2 above)
Comment (jeremycondon) [reply video/livestream-health-exports/proto/exportstream.proto:12]: Looks like what we need to do is nest the messages so you end up with:

```
message HealthUpdateEvent {
   message VideoTrack {
      optional uint64 stream_start_ms = 1;
      optional uint64 stream_end_ms = 2;
      optional uint64 bytes_received = 3;
      optional uint64 keyframes_received = 4;
      optional uint64 total_frames_received = 5;
    }
    required string environment_id = 1;
   required string live_stream_id = 2;
   required string asset_id = 3;
   repeated VideoTrack video_tracks = 7;
  ...
}

The google pubsub requirement is "one top-level type", we can have nested types

File: video/livestream-health-exports/proto/exportstream.proto [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+  optional uint64 total_frames_received = 5;
+}
+
+message CaptionTrack {
+  optional uint64 bytes_received = 1;
+  optional uint32 channel_count = 2;
+}
+
+message HealthUpdateEvent {
+  required string environment_id = 1;
```
Comment (jeremycondon) [inline video/livestream-health-exports/proto/exportstream.proto:30]: is this the internal environment_id or the customer facing environment_id?

**Pull request closed without merging**